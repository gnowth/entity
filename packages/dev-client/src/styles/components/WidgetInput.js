import { mixin } from '@gnowth/style';
import { css } from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const main = css`
  ${mixin({ name: 'componentBox' })}

  ${mixin({ name: 'disabled' })};
  ${mixin({ name: 'readOnly' })};
`;
