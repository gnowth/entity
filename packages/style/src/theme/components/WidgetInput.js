import { css } from 'styled-components';

import { mixin } from '../../selectors';

// eslint-disable-next-line import/prefer-default-export
export const component_widgetInput_standard = css`
  ${mixin({ name: 'componentBox' })}

  ${mixin({ name: 'disabled' })};
  ${mixin({ name: 'readOnly' })};
`;
