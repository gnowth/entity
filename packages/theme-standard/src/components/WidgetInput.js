import { mixins } from '@gnowth/theme';
import { css } from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const component_widgetInput_standard = {
  css: css`
    ${mixins.borderInput}
  `,
};
