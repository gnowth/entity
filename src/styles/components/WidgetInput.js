import { css } from 'styled-components';

export default css`
  ${props => props.theme.mixins.componentBox}

  ${props => props.disabled && props.theme.mixins.disabled};
  ${props => props.readOnly && props.theme.mixins.readOnly};
`;
