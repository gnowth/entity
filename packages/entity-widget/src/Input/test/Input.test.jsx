import renderer from 'react-test-renderer';
import React from 'react';
import { IntlProvider } from 'react-intl';

import WidgetInput from '..';

it('renders correctly', () => {
  const tree = renderer.create(
    <IntlProvider locale="en" messages={{}}>
      <WidgetInput
        name="test"
        onChange={() => undefined}
      />
    </IntlProvider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
