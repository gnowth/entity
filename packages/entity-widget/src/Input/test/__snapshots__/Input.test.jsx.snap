// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  width: 100%;
}

<input
  className="c0"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  name="test"
  namespace="component_widgetInput"
  variant="standard"
/>
`;
